src/index.js

```diff
@@ -480,7 +480,7 @@ if (nativeBindings.nativeOculusVR) {
     const layer = layers.find(layer => layer && layer.source && layer.source.tagName === 'CANVAS');
     if (layer) {
       const canvas = layer.source;
-      if (!vrPresentState.glContext) {
+      if (vrPresentState.glContext !== canvas._context) {
```

Comment:
Presuming this is correct, we'll probably need a similar condition for the other platforms:

- https://github.com/exokitxr/exokit/blob/e372292cf1caaa436862d2f8b3768a6804d9fa68/src/index.js#L601
- https://github.com/exokitxr/exokit/blob/e372292cf1caaa436862d2f8b3768a6804d9fa68/src/index.js#L768

As well as upcoming Oculus Go/Mobile.

Reply:
Looking at the below code I think this is the correct approach.

However, doing this naively will leak the previous presenting session and associated resources, such as the swap chain. We should take care to tear that down, or reuse it, or else there can be unpredictible effects.

The most correct way would be to tar down the session and re-make it, but [that is not implemented for the Oculus case](https://github.com/exokitxr/exokit/pull/931/files#diff-1fdf421c05c1140f6d71444ea2b27638L592).

Additionally, is is the intent to [clean up the way swap chains work](https://github.com/exokitxr/exokit/pull/929) which would affect this code path.

So maybe if we get this check implemented for all devices in this PR, we can open an issue for the resource cleanup?

Reply:
https://github.com/exokitxr/exokit/issues/932